extractor/src/main/java/org/schabi/newpipe/extractor/Image.java

```diff
@@ -0,0 +1,102 @@
+package org.schabi.newpipe.extractor;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+
+/**
+ * Class representing images in the extractor.
+ *
+ * <p>
+ * An image has three properties: its URL, its height and its width.
+ * </p>
+ *
+ * <p>
+ * Depending of the services, the height, the width or both properties may be not known.
+ * Implementations <b>must use</b> the relevant unknown constants in this case
+ * ({@link #HEIGHT_UNKNOWN} and {@link #WIDTH_UNKNOWN}), to ensure properly the lack of knowledge
+ * of one or both of these properties to extractor clients.
+ * </p>
+ */
+public final class Image implements Serializable {
+
+    /**
+     * Constant representing that the height of an {@link Image} is unknown.
+     */
+    public static final int HEIGHT_UNKNOWN = -1;
+
+    /**
+     * Constant representing that the width of an {@link Image} is unknown.
+     */
+    public static final int WIDTH_UNKNOWN = -1;
+
+    private final String url;
```

Comment:
The URL and its getters/setters should be annotated as either `@Nullable` or `@NonNull` to make spotting potential bugs easier.

Reply:
Done, should we enforce non-nullability and non-emptiness by throwing an `IllegalArgumentException` in the constructor if the URL is null or empty? I am not sure if that's a good idea or not.

Reply:
I don't think so. We don't do it at other places